<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<title>Unified Interface to spatial transcriptomics deconvolution tools • spacedeconv</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Unified Interface to spatial transcriptomics deconvolution tools">
<meta property="og:description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.0000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/spacedeconv.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/spacedeconv_data.html">Data Input</a>
    <a class="dropdown-item" href="articles/spacedeconv_example.html">Example</a>
    <a class="dropdown-item" href="articles/spacedeconv_faq.html">spacedeconv FAQ</a>
    <a class="dropdown-item" href="articles/spacedeconv_visualization.html">spacedeconv Visualization</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="spacedeconv">spacedeconv<a class="anchor" aria-label="anchor" href="#spacedeconv"></a>
</h1></div>

<p>spacedeconv is a unified interface to 31 deconvolution tools with focus on spatial transcriptomics datasets. In total 17 second-generation deconvolution tools are included, enabling deconvolution of any cell types when single-cell reference data is available. Additionally 10 first-generation tools, which are focusing on deconvolution of immune cells, are available as well as 4 first-generation methods optimised for mouse data.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>There are two ways to install <code>spacedeconv</code>:</p>
<ul>
<li>The <em>minimal</em> installation installs only the dependencies required for the basic functionalities. All deconvolution methods need to be installed on-demand.</li>
<li>The <em>complete</em> installation installs all dependencies including all deconvolution methods. This may take a considerable time.</li>
</ul>
<p>Since not all dependencies are on CRAN or Bioconductor, <code>spacedeconv</code>is available from GitHub only. We recommend installing trough the pak package manager:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the pak package manager</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># minimal installation</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="http://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># complete installation, including Python dependencies</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="http://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span>, dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># spacedeconv::install_all_python()</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>spacedeconv offers convenient access to perform first- and second-generation deconvolution on spatial transcriptomics datasets. While deconvolution can be performed directly with first-generation methods, second-generation algorithms require an additional annotated single-cell reference. A full list of deconvolution tools can be accessed by <code>spacedeconv::deconvolution_algorithms</code> or in the <a href="articles/spacedeconv_faq.html">FAQ</a>.</p>
<div class="section level3">
<h3 id="data-requirements">Data requirements<a class="anchor" aria-label="anchor" href="#data-requirements"></a>
</h3>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">SpatialExperiment</a></em>, will be deconvoluted</li>
<li>
<em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html" class="external-link">SingleCellExperiment</a></em> (recommended), <em><a href="https://anndata.dynverse.org/" class="external-link">anndata</a></em> or <em><a href="https://satijalab.org/seurat/" class="external-link">Seurat</a></em> containing cell type information</li>
</ul>
<p>The main functions of spacedeconv are used to build a signature matrix from annotated single-cell transcriptomics data and deconvolute spatially resolved transcriptomics datasets. The basic workflow consists of:</p>
</div>
<div class="section level3">
<h3 id="id_1-build-a-signature-matrix">1. Build a Signature Matrix<a class="anchor" aria-label="anchor" href="#id_1-build-a-signature-matrix"></a>
</h3>
<p>Build a cell type specific signature matrix from annotated single-cell reference data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  <span class="va">single_cell_object</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span>,</span>
<span>  assay_sc<span class="op">=</span><span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_2-deconvolution">2. Deconvolution<a class="anchor" aria-label="anchor" href="#id_2-deconvolution"></a>
</h3>
<p>To perform a deconvolution a SpatialExperiment object is required. Some methods additionally require a cell-type specific reference signature which can be calculated by <code><a href="reference/build_model.html">spacedeconv::build_model()</a></code>. By default the deconvolution results are added to the SpatialExperiment object to simplify the visualization. You can obtain the results in table form by setting <code>return_object=FALSE</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># save the results to an annotated SpatialExperiment</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spatial_object</span>,</span>
<span>  <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_3-visualization">3. Visualization<a class="anchor" aria-label="anchor" href="#id_3-visualization"></a>
</h3>
<p>spacedeconv includes multiple visualization functions. A full explanation of all visualization options can be found in this <a href="articles/spacedeconv_visualization.html">vignette</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># sample does refer to the first column of ColData(spe)</span></span>
<span><span class="co"># for cell_type input a celltype present in the deconvolution result</span></span>
<span><span class="fu"><a href="reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">spe</span>, cell_type<span class="op">=</span><span class="st">"spotlight_B.cells"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="additional-requirements">Additional Requirements<a class="anchor" aria-label="anchor" href="#additional-requirements"></a>
</h2>
<p>Most methods do not require additional software/tokens, but there are a few exceptions:</p>
<ul>
<li>A working version of Docker is required for CIBERSORTx</li>
<li>A token for CIBERSORTx is required from this website: <a href="https://cibersortx.stanford.edu/" class="external-link uri">https://cibersortx.stanford.edu/</a>
</li>
<li>The CIBERSORT source code is required for BSeq-sc (see tutorial in ?omnideconv::bseqsc_config)</li>
<li>SpatialExperiment requires <code>magick</code> to be installed: <code>sudo apt-get install libmagick++-dev</code>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="available-methods-licenses-citations">Available methods, Licenses, Citations<a class="anchor" aria-label="anchor" href="#available-methods-licenses-citations"></a>
</h2>
<p>Note that, while <em>spacedeconv</em> itself is free (<a href="https://github.com/omnideconv/omnideconv/blob/main/LICENSE" class="external-link">GPL 3.0</a>), you may need to obtain a license to use the individual methods. See the table below for more information. If you use this package in your work, please cite both our package and the method(s) you are using.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="13%">
<col width="53%">
</colgroup>
<thead><tr class="header">
<th>First-gen (immunedeconv)</th>
<th>First-gen mouse (immunedeconv)</th>
<th>Second-gen (omnideconv + spatial Methods)</th>
</tr></thead>
<tbody><tr class="odd">
<td></td>
<td></td>
<td></td>
</tr></tbody>
</table>
</div>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<table class="table">
<colgroup>
<col width="14%">
<col width="4%">
<col width="3%">
<col width="18%">
<col width="59%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th align="center">signature</th>
<th align="center">organism</th>
<th align="center">licence</th>
<th>citation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://github.com/omnideconv/omnideconv" class="external-link">omnideconv</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/omnideconv/omnideconv/blob/main/LICENSE" class="external-link">GPL-3</a></td>
<td>Citation will follow</td>
</tr>
<tr class="even">
<td><a href="https://github.com/omnideconv/immunedeconv" class="external-link">immunedeconv</a></td>
<td align="center"><span class="emoji" data-emoji="x">❌</span></td>
<td align="center">humand and mouse</td>
<td align="center"><a href="https://github.com/omnideconv/immunedeconv/blob/master/LICENSE" class="external-link">BSD</a></td>
<td>Sturm, G., Finotello, F., Petitprez, F., Zhang, J. D., Baumbach, J., Fridman, W. H., …, List, M., Aneichyk, T. (2019). Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology. Bioinformatics, 35(14), i436-i445.</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/RubD/Giotto/" class="external-link">spatialDWLS</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/RubD/Giotto/blob/master/LICENSE" class="external-link">MIT</a></td>
<td>Dong R, Yuan GC. SpatialDWLS: accurate deconvolution of spatial transcriptomic data. Genome Biol. 2021 May 10;22(1):145. doi: 10.1186/s13059-021-02362-7</td>
</tr>
<tr class="even">
<td><a href="https://github.com/BayraktarLab/cell2location" class="external-link">cell2location</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/BayraktarLab/cell2location/blob/master/LICENSE" class="external-link">Apache-2.0</a></td>
<td>Kleshchevnikov, V., Shmatko, A., Dann, E. et al. Cell2location maps fine-grained cell types in spatial transcriptomics. Nat Biotechnol (2022). <a href="https://doi.org/10.1038/s41587-021-01139-4" class="external-link uri">https://doi.org/10.1038/s41587-021-01139-4</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/MarcElosua/SPOTlight" class="external-link">SPOTlight</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/MarcElosua/SPOTlight/blob/main/LICENSE.md" class="external-link">GPL 3.0</a></td>
<td>Elosua-Bayes M, Nieto P, Mereu E, Gut I, Heyn H (2021): <em>SPOTlight: seeded NMF regression to deconvolute spatial transcriptomics spots with single-cell transcriptomes</em>. <strong>Nucleic Acids Res</strong> 49(9):e50. <a href="doi:10.1093/nar/gkab043" class="uri">doi:10.1093/nar/gkab043</a>.</td>
</tr>
<tr class="even">
<td><a href="https://github.com/dmcable/spacexr" class="external-link">RCTD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/dmcable/spacexr/blob/master/LICENSE" class="external-link">GPL 3.0</a></td>
<td>Cable, D.M., Murray, E., Zou, L.S. et al. Robust decomposition of cell type mixtures in spatial transcriptomics. Nat Biotechnol 40, 517–526 (2022). <a href="https://doi.org/10.1038/s41587-021-00830-w" class="external-link uri">https://doi.org/10.1038/s41587-021-00830-w</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/YingMa0107/CARD" class="external-link">CARD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">✔️</span></td>
<td align="center">human</td>
<td align="center"><a href="https://github.com/YingMa0107/CARD/blob/master/LICENSE.md" class="external-link">GPL-3.0</a></td>
<td>Ying Ma, and Xiang Zhou (2022). Spatially informed cell type deconvolution for spatial transcriptomics. <a href="https://doi.org/10.1038/s41587-022-01273-7" class="external-link uri">https://doi.org/10.1038/s41587-022-01273-7</a>
</td>
</tr>
</tbody>
</table>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing spacedeconv</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Constantin Zackl <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/omnideconv/spacedeconv/actions/workflows/test.yml" class="external-link"><img src="https://github.com/omnideconv/spacedeconv/actions/workflows/test.yml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://github.com/omnideconv/spacedeconv/actions/workflows/pkgdown.yml" class="external-link"><img src="https://github.com/omnideconv/spacedeconv/actions/workflows/pkgdown.yml/badge.svg" alt="docs"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
