<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="spacedeconv">
<title>Example • spacedeconv</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Example">
<meta property="og:description" content="spacedeconv">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.0000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/spacedeconv.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/spacedeconv_colocalization.html">Colocalization</a>
    <a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a>
    <a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a>
    <a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a>
    <a class="dropdown-item" href="../articles/spacedeconv_metacell.html">Metacell</a>
    <a class="dropdown-item" href="../articles/spacedeconv_visualization.html">spacedeconv Visualization</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Example</h1>
            
      
      
      <div class="d-none name"><code>spacedeconv_example.Rmd</code></div>
    </div>

    
    
<p>spacedeconv is a unified interface for the deconvolution of spatial
transcriptomics data. In total spacedeconv gives access to 31
deconvolution methods:</p>
<pre><code><span><span class="co">#&gt;             RCTD        SPOTlight             CARD      spatialDWLS </span></span>
<span><span class="co">#&gt;           "rctd"      "spotlight"           "card"    "spatialdwls" </span></span>
<span><span class="co">#&gt;    cell2location        AutoGeneS       BayesPrism           Bisque </span></span>
<span><span class="co">#&gt;  "cell2location"      "autogenes"     "bayesprism"         "bisque" </span></span>
<span><span class="co">#&gt;          BSeq-sc       CIBERSORTx            CDSeq              CPM </span></span>
<span><span class="co">#&gt;         "bseqsc"     "cibersortx"          "cdseq"            "cpm" </span></span>
<span><span class="co">#&gt;             DWLS             MOMF            MuSiC           Scaden </span></span>
<span><span class="co">#&gt;           "dwls"           "momf"          "music"         "scaden" </span></span>
<span><span class="co">#&gt;             SCDC       MCPcounter             EPIC        quanTIseq </span></span>
<span><span class="co">#&gt;           "scdc"    "mcp_counter"           "epic"      "quantiseq" </span></span>
<span><span class="co">#&gt;            xCell        CIBERSORT CIBERSORT (abs.)            TIMER </span></span>
<span><span class="co">#&gt;          "xcell"      "cibersort"  "cibersort_abs"          "timer" </span></span>
<span><span class="co">#&gt;     ConsensusTME             ABIS         ESTIMATE      mMCPcounter </span></span>
<span><span class="co">#&gt;  "consensus_tme"           "abis"       "estimate"   "mmcp_counter" </span></span>
<span><span class="co">#&gt;        seqImmuCC              DCQ             BASE </span></span>
<span><span class="co">#&gt;      "seqimmucc"            "dcq"           "base"</span></span></code></pre>
<div class="section level2">
<h2 id="sample-data">Sample Data<a class="anchor" aria-label="anchor" href="#sample-data"></a>
</h2>
<p>For this tutorial we will use one of spacedeconvs sample dataset and
the deconvolution algorithm “SCDC”.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">spacedeconv</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing-and-normalization">Preprocessing and Normalization<a class="anchor" aria-label="anchor" href="#preprocessing-and-normalization"></a>
</h2>
<p>To get a first insight into the data we type the dataset name into
the terminal:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 29733 7986 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(29733): RP11-34P13.7 FO538757.3 ... KRTAP9-2 IGLVIV-66-1</span></span>
<span><span class="co">#&gt; rowData names(1): ID</span></span>
<span><span class="co">#&gt; colnames(7986): CID44971_AAGCCGCCACGCATCG CID44971_AAGGAGCTCAACACAC ...</span></span>
<span><span class="co">#&gt;   CID44971_TTTCCTCCAAGCCATT CID44971_TTTGGTTGTATCACCA</span></span>
<span><span class="co">#&gt; colData names(10): Sample Barcode ... celltype_minor celltype_major</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>We can see that the single-cell data is available as a <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a>.
In total this dataset includes expression values from 29733 genes and
7986 cells. It can further be seen that cell type information is
available in <code>colData</code> where cell related annotation is
stored.</p>
<p>As the next step we normalize the data as <code>cpm</code> (“Counts
Per Million”)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; Removing 8 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; Removing 5862 variables with all zero expression</span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; Removing 137 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; Removing 13049 variables with all zero expression</span></span>
<span></span>
<span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">single_cell_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Normalized object using cpm. </span></span>
<span><span class="co">#&gt; Note that the normalization is saved in an additional assay.</span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spatial_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Normalized object using cpm. </span></span>
<span><span class="co">#&gt; Note that the normalization is saved in an additional assay.</span></span></code></pre></div>
<p>When we access the object again we can see that the normalization has
been added to the object in an additional assay called
<code>"cpm"</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 23871 7978 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(2): counts cpm</span></span>
<span><span class="co">#&gt; rownames(23871): RP11-34P13.7 FO538757.3 ... LINC01425 AP000477.2</span></span>
<span><span class="co">#&gt; rowData names(1): ID</span></span>
<span><span class="co">#&gt; colnames(7978): CID44971_AAGCCGCCACGCATCG CID44971_AAGGAGCTCAACACAC ...</span></span>
<span><span class="co">#&gt;   CID44971_TTTCCTCCAAGCCATT CID44971_TTTGGTTGTATCACCA</span></span>
<span><span class="co">#&gt; colData names(10): Sample Barcode ... celltype_minor celltype_major</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-model">Build Model<a class="anchor" aria-label="anchor" href="#build-model"></a>
</h2>
<p>To build a model we need to provide a single cell expression object
annotated with cell type information. Let’s have a look into the object
to see which annotation is available.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Sample"          "Barcode"         "orig.ident"      "nCount_RNA"     </span></span>
<span><span class="co">#&gt;  [5] "nFeature_RNA"    "percent.mito"    "subtype"         "celltype_subset"</span></span>
<span><span class="co">#&gt;  [9] "celltype_minor"  "celltype_major"</span></span></code></pre></div>
<p>There are three columns with cell_type information. We will inspect
these columns further to decide which one we are going to use for
signature calculation.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># celtype_subset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">$</span><span class="va">celltype_subset</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Endothelial ACKR1"              "Endothelial RGS5"              </span></span>
<span><span class="co">#&gt;  [3] "Endothelial CXCL12"             "CAFs MSC iCAF-like s1"         </span></span>
<span><span class="co">#&gt;  [5] "CAFs MSC iCAF-like s2"          "CAFs Transitioning s3"         </span></span>
<span><span class="co">#&gt;  [7] "CAFs myCAF like s4"             "CAFs myCAF like s5"            </span></span>
<span><span class="co">#&gt;  [9] "PVL Differentiated s3"          "PVL_Immature s2"               </span></span>
<span><span class="co">#&gt; [11] "PVL Immature s1"                "Endothelial Lymphatic LYVE1"   </span></span>
<span><span class="co">#&gt; [13] "B cells Memory"                 "B cells Naive"                 </span></span>
<span><span class="co">#&gt; [15] "Plasmablasts"                   "T_cells_c4_CD8+_ZFP36"         </span></span>
<span><span class="co">#&gt; [17] "T_cells_c6_IFIT1"               "T_cells_c5_CD8+_GZMK"          </span></span>
<span><span class="co">#&gt; [19] "T_cells_c7_CD8+_IFNG"           "T_cells_c8_CD8+_LAG3"          </span></span>
<span><span class="co">#&gt; [21] "T_cells_c0_CD4+_CCR7"           "T_cells_c1_CD4+_IL7R"          </span></span>
<span><span class="co">#&gt; [23] "T_cells_c2_CD4+_T-regs_FOXP3"   "T_cells_c3_CD4+_Tfh_CXCL13"    </span></span>
<span><span class="co">#&gt; [25] "T_cells_c9_NK_cells_AREG"       "T_cells_c11_MKI67"             </span></span>
<span><span class="co">#&gt; [27] "T_cells_c10_NKT_cells_FCGR3A"   "Myeloid_c10_Macrophage_1_EGR1" </span></span>
<span><span class="co">#&gt; [29] "Myeloid_c12_Monocyte_1_IL1B"    "Myeloid_c2_LAM2_APOE"          </span></span>
<span><span class="co">#&gt; [31] "Myeloid_c1_LAM1_FABP5"          "Myeloid_c8_Monocyte_2_S100A9"  </span></span>
<span><span class="co">#&gt; [33] "Myeloid_c7_Monocyte_3_FCGR3A"   "Myeloid_c9_Macrophage_2_CXCL10"</span></span>
<span><span class="co">#&gt; [35] "Cycling_Myeloid"                "Myeloid_c11_cDC2_CD1C"         </span></span>
<span><span class="co">#&gt; [37] "Myeloid_c4_DCs_pDC_IRF7"        "Myeloid_c3_cDC1_CLEC9A"        </span></span>
<span><span class="co">#&gt; [39] "Myeloid_c0_DC_LAMP3"            "Cancer LumB SC"                </span></span>
<span><span class="co">#&gt; [41] "Cancer Cycling"                 "Cancer Basal SC"               </span></span>
<span><span class="co">#&gt; [43] "Cancer Her2 SC"                 "Myoepithelial"                 </span></span>
<span><span class="co">#&gt; [45] "Luminal Progenitors"            "Mature Luminal"</span></span>
<span></span>
<span><span class="co"># celltype_minor</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">$</span><span class="va">celltype_minor</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Endothelial ACKR1"           "Endothelial RGS5"           </span></span>
<span><span class="co">#&gt;  [3] "Endothelial CXCL12"          "CAFs MSC iCAF-like"         </span></span>
<span><span class="co">#&gt;  [5] "CAFs myCAF-like"             "PVL Differentiated"         </span></span>
<span><span class="co">#&gt;  [7] "PVL Immature"                "Endothelial Lymphatic LYVE1"</span></span>
<span><span class="co">#&gt;  [9] "B cells Memory"              "B cells Naive"              </span></span>
<span><span class="co">#&gt; [11] "Plasmablasts"                "T cells CD8+"               </span></span>
<span><span class="co">#&gt; [13] "T cells CD4+"                "NK cells"                   </span></span>
<span><span class="co">#&gt; [15] "Cycling T-cells"             "NKT cells"                  </span></span>
<span><span class="co">#&gt; [17] "Macrophage"                  "Monocyte"                   </span></span>
<span><span class="co">#&gt; [19] "Cycling_Myeloid"             "DCs"                        </span></span>
<span><span class="co">#&gt; [21] "Cancer LumB SC"              "Cancer Cycling"             </span></span>
<span><span class="co">#&gt; [23] "Cancer Basal SC"             "Cancer Her2 SC"             </span></span>
<span><span class="co">#&gt; [25] "Myoepithelial"               "Luminal Progenitors"        </span></span>
<span><span class="co">#&gt; [27] "Mature Luminal"</span></span>
<span></span>
<span><span class="co"># celltype_major</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">$</span><span class="va">celltype_major</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Endothelial"       "CAFs"              "PVL"              </span></span>
<span><span class="co">#&gt; [4] "B-cells"           "Plasmablasts"      "T-cells"          </span></span>
<span><span class="co">#&gt; [7] "Myeloid"           "Cancer Epithelial" "Normal Epithelial"</span></span></code></pre></div>
<p>There are three different resolutions of cell type annotation
available. For this tutorial we will choose
<code>"celltype_major"</code> as cell type annotation.</p>
<p>Next we can calculate a signature using the annotated single-data and
the deconvolution tool “SCDC”.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"estimate"</span>,</span>
<span>  spatial_obj <span class="op">=</span> <span class="va">spatial_data_3</span>,</span>
<span>  assay_sc <span class="op">=</span> <span class="st">"cpm"</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span>,</span>
<span>  batch_id_col <span class="op">=</span> <span class="st">"orig.ident"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The signature contains expression values for all celltypes and a
subset of distinctly expressed genes. The following example shows the
first 10 rows of the signature calculated in the previous step.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># knitr::kable(round(signature[1:10, ], 4))</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="deconvolution">Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h2>
<p>To deconvolute the Visium slide we use the signature we just
calculated. Depending on the selected deconvolution tool further
parameters need to be passed, in this case <code>cell_type_col</code>
and <code>batch_id_col</code> as well as the scRNA-seq reference
dataset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconv</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spatial_data_3</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"estimate"</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span>,</span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_minor"</span>,</span>
<span>  batch_id_col <span class="op">=</span> <span class="st">"orig.ident"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Merged dataset includes 9352 genes (1060 mismatched)."</span></span>
<span><span class="co">#&gt; [1] "1 gene set: StromalSignature  overlap= 135"</span></span>
<span><span class="co">#&gt; [1] "2 gene set: ImmuneSignature  overlap= 139"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The deconvolution results are stored in the spatial object to
simplify visualizations. spacedeconv offers 6 visualization functions
with different scopes, further explained in the visualization <a href="spacedeconv_visualization.html">vignette</a>.</p>
<p>We can see the available deconvolution results by using
<code>available_results(deconv)</code> or by accessing the objects
<code>colData</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/available_results.html">available_results</a></span><span class="op">(</span><span class="va">deconv</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "estimate_stroma.score"   "estimate_immune.score"  </span></span>
<span><span class="co">#&gt; [3] "estimate_estimate.score" "estimate_tumor.purity"</span></span></code></pre></div>
<p>For each Spot the estimated cell fraction is available as object
annotation and can be used for visualizations.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"estimate_immune.score"</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_example_files/figure-html/visualization-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
