<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="spacedeconv">
<title>FAQ • spacedeconv</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FAQ">
<meta property="og:description" content="spacedeconv">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.0000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/spacedeconv.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a>
    <a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a>
    <a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a>
    <a class="dropdown-item" href="../articles/spacedeconv_visualization.html">spacedeconv Visualization</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>FAQ</h1>
            
      
      
      <div class="d-none name"><code>spacedeconv_faq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="faq">FAQ<a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<div class="section level4">
<h4 id="my-spatialexperiment-contains-ensembl-ids--can-i-still-perform-deconvolution">My SpatialExperiment contains ENSEMBL IDs. Can i still perform
deconvolution?<a class="anchor" aria-label="anchor" href="#my-spatialexperiment-contains-ensembl-ids--can-i-still-perform-deconvolution"></a>
</h4>
<p>It is possible to perform second-generation deconvolution algorithms
when annotated scRNA-seq data containing ENSEMBL IDs is available. All
first-gen deconvolution tools require HGNC symbols. You may also check
your SpatialExperiment <code>rowData()</code> for available annotation
to overwrite ENSEMBL IDs. In most cases this can be solved by using the
following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="how-can-i-see-which-deconvolution-results-are-available-in-my-object">How can i see which deconvolution results are available in my
object?<a class="anchor" aria-label="anchor" href="#how-can-i-see-which-deconvolution-results-are-available-in-my-object"></a>
</h4>
<p>You can use <code>available_results(object)</code> or access
<code>names(colData(object))</code> to see which deconvolution results
are available.</p>
</div>
<div class="section level4">
<h4 id="the-spacedeconv-plots-contain-a-lot-of-white-spots">The spacedeconv plots contain a lot of white spots<a class="anchor" aria-label="anchor" href="#the-spacedeconv-plots-contain-a-lot-of-white-spots"></a>
</h4>
<p>When reading raw spatial data the SpatialExperiment contains
expression information for all spots, even if they are not located over
tissue. Remove those spots with the following code</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="the-visium-slide-and-the-spatial-data-are-not-rotated-in-the-same-direction">The Visium Slide and the spatial data are not rotated in the same
direction<a class="anchor" aria-label="anchor" href="#the-visium-slide-and-the-spatial-data-are-not-rotated-in-the-same-direction"></a>
</h4>
<p>The SpatialExperiment wrapper includes function to rotate/mirror the
image. More information can be found in the Visualization Vignette or
the SpatialExperiment Documentation.</p>
</div>
<div class="section level4">
<h4 id="the-hexagons-in-my-plots-are-rotated-and-dont-give-an-even-appearance">The hexagons in my plots are rotated and don’t give an even
appearance<a class="anchor" aria-label="anchor" href="#the-hexagons-in-my-plots-are-rotated-and-dont-give-an-even-appearance"></a>
</h4>
<p>Your Visium slide might be aligned incorrectly. Try rotating the
image or use <code>offset_rotation=TRUE</code> as a parameter in the
plotting function.</p>
</div>
<div class="section level4">
<h4 id="which-deconvolution-tools-are-available">Which deconvolution tools are available?<a class="anchor" aria-label="anchor" href="#which-deconvolution-tools-are-available"></a>
</h4>
<p>This list includes the methods name and the internally used tokens.
More detailed descriptions of each method can be found at the end of
this vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="va"><a href="../reference/deconvolution_methods.html">deconvolution_methods</a></span></span></code></pre></div>
<pre><code><span><span class="co">##             RCTD        SPOTlight             CARD      spatialDWLS </span></span>
<span><span class="co">##           "rctd"      "spotlight"           "card"    "spatialdwls" </span></span>
<span><span class="co">##    cell2location        AutoGeneS       BayesPrism           Bisque </span></span>
<span><span class="co">##  "cell2location"      "autogenes"     "bayesprism"         "bisque" </span></span>
<span><span class="co">##          BSeq-sc       CIBERSORTx            CDSeq              CPM </span></span>
<span><span class="co">##         "bseqsc"     "cibersortx"          "cdseq"            "cpm" </span></span>
<span><span class="co">##             DWLS             MOMF            MuSiC           Scaden </span></span>
<span><span class="co">##           "dwls"           "momf"          "music"         "scaden" </span></span>
<span><span class="co">##             SCDC       MCPcounter             EPIC        quanTIseq </span></span>
<span><span class="co">##           "scdc"    "mcp_counter"           "epic"      "quantiseq" </span></span>
<span><span class="co">##            xCell        CIBERSORT CIBERSORT (abs.)            TIMER </span></span>
<span><span class="co">##          "xcell"      "cibersort"  "cibersort_abs"          "timer" </span></span>
<span><span class="co">##     ConsensusTME             ABIS         ESTIMATE      mMCPcounter </span></span>
<span><span class="co">##  "consensus_tme"           "abis"       "estimate"   "mmcp_counter" </span></span>
<span><span class="co">##        seqImmuCC              DCQ             BASE </span></span>
<span><span class="co">##      "seqimmucc"            "dcq"           "base"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="method-recommendations">Method recommendations<a class="anchor" aria-label="anchor" href="#method-recommendations"></a>
</h2>
<div class="section level3">
<h3 id="autogenes">AutoGeneS<a class="anchor" aria-label="anchor" href="#autogenes"></a>
</h3>
<p>AutoGeneS <span class="citation">(Aliee and Theis 2020)</span>
selects marker genes based on a multi-objective optimization which
simultaneously minimizes the correlation and maximizes the distance
between cell-type signatures. Compared to approaches selecting genes
only on differential expression it performs favorably with respect to
closely related cell-types. The resulting signature matrix can be
combined with different regression models out-of-the-box, including
𝜈-support vector regression (SVR).</p>
</div>
<div class="section level3">
<h3 id="bisque">Bisque<a class="anchor" aria-label="anchor" href="#bisque"></a>
</h3>
<p>Bisque <span class="citation">(Jew et al. 2020)</span> takes
advantage of matched bulk and scRNA-seq samples to improve accuracy of
cell-type estimates. Using linear regression, it corrects for
platform-specific biases between pseudo-bulk reference profiles derived
from scRNA-seq and bulk RNA-seq samples. It then applies NNLS-regression
to deconvolute other bulk RNA-seq samples.</p>
</div>
<div class="section level3">
<h3 id="bseq-sc">BSeq-sc<a class="anchor" aria-label="anchor" href="#bseq-sc"></a>
</h3>
<p>Bseq-SC <span class="citation">(Baron et al. 2016)</span> builds
signature matrices by averaging a manually defined list of marker genes
across cell-type-specific single-cell gene expression profiles. It
leverages the CIBERSORT algorithm(Newman et al., 2015) to perform
deconvolution.</p>
</div>
<div class="section level3">
<h3 id="cdseq">CDSeq<a class="anchor" aria-label="anchor" href="#cdseq"></a>
</h3>
<p>“Here we present an R implementation of CDSeq (CDSeqR) with
significant performance improvement over the original implementation in
MATLAB and an added new function to aid cell type annotation. We
developed a novel strategy to substantially improve computational
efficiency in both speed and memory usage. In addition, we designed and
implemented a new function for annotating the CDSeq estimated cell types
using single-cell RNA sequencing (scRNA-seq) data. This function allows
users to readily interpret and visualize the CDSeq estimated cell types.
In addition, this new function further allows the users to annotate
CDSeq-estimated cell types using marker genes.” From CDSeqR paper <span class="citation">(Kang et al. 2021)</span>. Note: it performs also
complete deconvolution.</p>
</div>
<div class="section level3">
<h3 id="cibersortx">CIBERSORTx<a class="anchor" aria-label="anchor" href="#cibersortx"></a>
</h3>
<p>CIBERSORTx <span class="citation">(A. M. Newman et al. 2019)</span>
is an extension to the original CIBERSORT algorithm <span class="citation">(A. M. et al. Newman 2015)</span> that enables building
signature matrices from scRNA-seq or sorted bulk RNA-seq samples based
on differential gene expression analysis. To address technical biases
they introduce two batch correction methods, one designed to mitigate
batch effects between datasets, the other to address differences between
sequencing protocols (e.g. 10x vs. full-length). Moreover, CIBERSORTx
supports “complete” deconvolution, not only yielding cell-type
proportions but disentangling bulk gene expression profiles into
cell-type specific expression profiles.</p>
</div>
<div class="section level3">
<h3 id="cpm">CPM<a class="anchor" aria-label="anchor" href="#cpm"></a>
</h3>
<p>Cell Population Mapping (CPM) <span class="citation">(Frishberg et
al. 2019)</span> works independent of discrete cell-types. Instead of
building a cell-type specific signature matrix, it directly uses the
single-cell gene expression as reference and performs linear SVR to
predict relative abundances of cell-states organized in a continuous
1-dimensional manifold or 2-dimensional space. To increase robustness,
the results are averaged over independent predictions on random subsets
of the single-cell RNA-seq dataset and smoothed over a neighborhood of
the 10 nearest reference cells.</p>
</div>
<div class="section level3">
<h3 id="dwls">DWLS<a class="anchor" aria-label="anchor" href="#dwls"></a>
</h3>
<p>Tsoucas et al. introduce dampened weighted least squares
(DWLS)-regression <span class="citation">(Tsoucas et al. 2019)</span>,
which improves over ordinary least squares regression or 𝜈-SVR in that
it attributes higher weight to rare cell types and marker genes with a
low average expression level. They show that their method achieves
favorable performance when estimating the abundance of infrequent
cell-types.</p>
</div>
<div class="section level3">
<h3 id="momf">MOMF<a class="anchor" aria-label="anchor" href="#momf"></a>
</h3>
<p>With MOMF <span class="citation">(Sun, Sun, and Yang 2019)</span>,
Multi-Omics Matrix Factorization, models the cell type specific mean in
scRNA-seq data by also accounting for the uncertainty of these cell type
specific mean gene expression levels. Alternating Direction Method of
Multipliers algorithm is then used to estimate the parameters in bulk
RNA-seq downstream analyses.</p>
</div>
<div class="section level3">
<h3 id="music">MuSiC<a class="anchor" aria-label="anchor" href="#music"></a>
</h3>
<p>MuSiC <span class="citation">(Wang et al. 2019)</span> does not
pre-select marker genes but rather weights marker genes from low to high
variance between subjects. It then performs non-negative least squares
(NNLS)-regression to obtain cell-type estimates. To improve performance
on closely related cell-types MuSiC incorporates hierarchical
information from unsupervised clustering: It first deconvolves a gene
expression profile into coarse cell-type clusters and subsequently uses
more specific marker genes to estimate subtypes within these clusters. A
recent extension of the MuSiC framework(Sosina et al., n.d.) addresses
different mRNA contents of different cell-types, enabling MuSiC to
generate absolute scores that can be compared both between samples and
cell-types.</p>
</div>
<div class="section level3">
<h3 id="scaden">Scaden<a class="anchor" aria-label="anchor" href="#scaden"></a>
</h3>
<p>Scaden <span class="citation">(Menden et al. 2020)</span> leverages a
deep neural network (DNN) for estimating cell-type proportions. Instead
of explicitly building a signature matrix, the DNN implicitly learns
which features are important for a certain cell type. The DNN is trained
by simulating bulk RNA-seq samples with known cell-type proportions from
scRNA-seq datasets. To increase robustness, the training process allows
to flexibly integrate multiple scRNA-seq datasets and, optionally, bulk
RNA-seq samples with “gold standard” measurements such as FACS.</p>
</div>
<div class="section level3">
<h3 id="scdc">SCDC<a class="anchor" aria-label="anchor" href="#scdc"></a>
</h3>
<p>SCDC <span class="citation">(Dong et al. 2020)</span> is an extension
to the MuSiC framework(Wang et al., 2019). It adds an additional scaling
step to down-weight the impact of aberrant genes on the prediction and a
filtering strategy for misclassified cells and doublets. Most
importantly, it adds an Ensemble layer allowing for a combined
prediction on multiple, independent scRNA-seq datasets. The latter
implicitly addresses batch effects between datasets and has been shown
to improve robustness and overall performance.</p>
</div>
<div class="section level3">
<h3 id="computation-time-estimates">Computation Time Estimates<a class="anchor" aria-label="anchor" href="#computation-time-estimates"></a>
</h3>
<p>For this example each method deconvoluted the sample dataset with
ending “_1”. The computation times can be used to select suitable
deconvolution methods.</p>
<table class="table">
<thead><tr class="header">
<th>Method</th>
<th>Computation time in sec</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Estimate</td>
<td>9.894</td>
</tr>
<tr class="even">
<td>RCTD</td>
<td>511.137</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Aliee2020" class="csl-entry">
Aliee, Hananeh, and Fabian Theis. 2020. <span>“AutoGeneS: Automatic Gene
Selection Using Multi-Objective Optimization for RNA-Seq
Deconvolution.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2020.02.21.940650" class="external-link">https://doi.org/10.1101/2020.02.21.940650</a>.
</div>
<div id="ref-Baron2016" class="csl-entry">
Baron, Maayan, Adrian Veres, Samuel L. Wolock, Aubrey L. Faust, Renaud
Gaujoux, Amedeo Vetere, Jennifer Hyoje Ryu, et al. 2016. <span>“A
Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals
Inter- and Intra-Cell Population Structure.”</span> <em>Cell
Systems</em> 3 (4): 346–360.e4. <a href="https://doi.org/10.1016/j.cels.2016.08.011" class="external-link">https://doi.org/10.1016/j.cels.2016.08.011</a>.
</div>
<div id="ref-Dong2020" class="csl-entry">
Dong, Meichen, Aatish Thennavan, Eugene Urrutia, Yun Li, Charles M
Perou, Fei Zou, and Yuchao Jiang. 2020. <span>“<span class="nocase">SCDC: bulk gene expression deconvolution by multiple
single-cell RNA sequencing references</span>.”</span> <em>Briefings in
Bioinformatics</em> 22 (1): 416–27. <a href="https://doi.org/10.1093/bib/bbz166" class="external-link">https://doi.org/10.1093/bib/bbz166</a>.
</div>
<div id="ref-Frishberg2019" class="csl-entry">
Frishberg, Amit, Naama Peshes-Yaloz, Ofir Cohn, Diana Rosentul, Yael
Steuerman, Liran Valadarsky, Gal Yankovitz, et al. 2019. <span>“Cell
Composition Analysis of Bulk Genomics Using Single-Cell Data.”</span>
<em>Nature Methods</em> 16 (4): 327–32. <a href="https://doi.org/10.1038/s41592-019-0355-5" class="external-link">https://doi.org/10.1038/s41592-019-0355-5</a>.
</div>
<div id="ref-Jew2020" class="csl-entry">
Jew, Brandon, Marcus Alvarez, Elior Rahmani, Zong Miao, Arthur Ko,
Kristina M. Garske, Jae Hoon Sul, Kirsi H. Pietiläinen, Päivi Pajukanta,
and Eran Halperin. 2020. <span>“Accurate Estimation of Cell Composition
in Bulk Expression Through Robust Integration of Single-Cell
Information.”</span> <em>Nature Communications</em> 11 (1): 1971. <a href="https://doi.org/10.1038/s41467-020-15816-6" class="external-link">https://doi.org/10.1038/s41467-020-15816-6</a>.
</div>
<div id="ref-Kang2021" class="csl-entry">
Kang, Kai, Caizhi Huang, Yuanyuan Li, David M. Umbach, and Leping Li.
2021. <span>“CDSeqR: Fast Complete Deconvolution for Gene Expression
Data from Bulk Tissues.”</span> <em>BMC Bioinformatics</em> 22 (1): 262.
<a href="https://doi.org/10.1186/s12859-021-04186-5" class="external-link">https://doi.org/10.1186/s12859-021-04186-5</a>.
</div>
<div id="ref-Mendeneaba2619" class="csl-entry">
Menden, Kevin, Mohamed Marouf, Sergio Oller, Anupriya Dalmia, Daniel
Sumner Magruder, Karin Kloiber, Peter Heutink, and Stefan Bonn. 2020.
<span>“Deep Learning<span></span>based Cell Composition Analysis from
Tissue Expression Profiles.”</span> <em>Science Advances</em> 6 (30). <a href="https://doi.org/10.1126/sciadv.aba2619" class="external-link">https://doi.org/10.1126/sciadv.aba2619</a>.
</div>
<div id="ref-Newman2015" class="csl-entry">
Newman, Aaron M. et al. 2015. <span>“<span class="nocase">Robust
enumeration of cell subsets from tissue expression
profiles</span>.”</span> <em>Nature Methods</em> 12. <a href="https://doi.org/10.1038/nmeth.3337" class="external-link">https://doi.org/10.1038/nmeth.3337</a>.
</div>
<div id="ref-Newman2019" class="csl-entry">
Newman, Aaron M., Chloé B. Steen, Chih Long Liu, Andrew J. Gentles,
Aadel A. Chaudhuri, Florian Scherer, Michael S. Khodadoust, et al. 2019.
<span>“Determining Cell Type Abundance and Expression from Bulk Tissues
with Digital Cytometry.”</span> <em>Nature Biotechnology</em> 37 (7):
773–82. <a href="https://doi.org/10.1038/s41587-019-0114-2" class="external-link">https://doi.org/10.1038/s41587-019-0114-2</a>.
</div>
<div id="ref-Sun2019" class="csl-entry">
Sun, X., S. Sun, and S. Yang. 2019. <span>“An Efficient and Flexible
Method for Deconvoluting Bulk RNA-Seq Data with Single-Cell RNA-Seq
Data.”</span> <em>Cells</em> 8 (10). <a href="https://doi.org/10.3390/cells8101161" class="external-link">https://doi.org/10.3390/cells8101161</a>.
</div>
<div id="ref-Tsoucas2019" class="csl-entry">
Tsoucas, Daphne, Rui Dong, Haide Chen, Qian Zhu, Guoji Guo, and
Guo-Cheng Yuan. 2019. <span>“Accurate Estimation of Cell-Type
Composition from Gene Expression Data.”</span> <em>Nature
Communications</em> 10 (1): 2975. <a href="https://doi.org/10.1038/s41467-019-10802-z" class="external-link">https://doi.org/10.1038/s41467-019-10802-z</a>.
</div>
<div id="ref-Wang2019" class="csl-entry">
Wang, Xuran, Jihwan Park, Katalin Susztak, Nancy R. Zhang, and Mingyao
Li. 2019. <span>“Bulk Tissue Cell Type Deconvolution with Multi-Subject
Single-Cell Expression Reference.”</span> <em>Nature Communications</em>
10 (1): 380. <a href="https://doi.org/10.1038/s41467-018-08023-x" class="external-link">https://doi.org/10.1038/s41467-018-08023-x</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
